File: problems/0009_palindrome-number/to-string/Solution.php
<?php
class Solution {
    /**
     * @param Integer $x
     * @return Boolean
     */
    function isPalindrome($x) {
        if ($x < 0)
            return false;
        else if ($x < 10)
            return true;
        $s = strval($x);
        for ($i = 0, $j = strlen($s) - 1; $i < $j; ++$i, --$j)
            if ($s[$i] != $s[$j])
                return false;
        return true;
    }
}

File: problems/0009_palindrome-number/to-string/Solution.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0)
            return false;
        else if (x < 10)
            return true;
        final String s = String.valueOf(x);
        for (int i = 0, j = s.length() - 1; i < j; ++i, --j)
            if (s.charAt(i) != s.charAt(j))
                return false;
        return true;
    }
}
